core/src/test/java/com/linecorp/armeria/server/conditional/ConditionalRequestUtilTest.java

```diff
@@ -0,0 +1,145 @@
+                .isTrue();
+    }
+
+    @Test
+    void testStrongComparison() {
+        assertThat(strongComparison(ImmutableList.of(new ETag("1", true)), new ETag("1", true)))
+                .isFalse();
+        assertThat(strongComparison(ImmutableList.of(new ETag("1", true)), new ETag("2", true)))
+                .isFalse();
+        assertThat(strongComparison(ImmutableList.of(new ETag("1", false)), new ETag("1", true)))
+                .isFalse();
+        assertThat(strongComparison(ImmutableList.of(new ETag("1", false)), new ETag("1", false)))
+                .isTrue();
+    }
+
+    @Test
+    void testIfMatch() {
+        assertThat(ConditionalRequestUtil.ifMatch("W/\"1\"", new ETag("1", true)))
+                .isSameAs(ETagResponse.PERFORM_METHOD);
+        assertThat(ConditionalRequestUtil.ifMatch("*", new ETag("1", true)))
+                .isSameAs(ETagResponse.PERFORM_METHOD);
+        assertThat(ConditionalRequestUtil.ifMatch("*", null))
+                .isSameAs(ETagResponse.SKIP_METHOD_PRECONDITION_FAILED);
+        assertThat(ConditionalRequestUtil.ifMatch("\"1\"", new ETag("1", true)))
+                .isSameAs(ETagResponse.PERFORM_METHOD);
+        assertThat(ConditionalRequestUtil.ifMatch("\"1\"", new ETag("1", false)))
+                .isSameAs(ETagResponse.SKIP_METHOD_PRECONDITION_FAILED);
+    }
+
+    @Test
+    void testIfNoneMatch() {
+        assertThat(ConditionalRequestUtil.ifNoneMatch("W/\"1\"", new ETag("1", true)))
+                .isSameAs(ETagResponse.SKIP_METHOD_NOT_MODIFIED);
+        assertThat(ConditionalRequestUtil.ifNoneMatch("*", new ETag("1", true)))
+                .isSameAs(ETagResponse.SKIP_METHOD_NOT_MODIFIED);
+        assertThat(ConditionalRequestUtil.ifNoneMatch("*", null))
+                .isSameAs(ETagResponse.PERFORM_METHOD);
+        assertThat(ConditionalRequestUtil.ifNoneMatch("\"1\"", new ETag("1", true)))
+                .isSameAs(ETagResponse.SKIP_METHOD_NOT_MODIFIED);
+        assertThat(ConditionalRequestUtil.ifNoneMatch("\"1\"", new ETag("1", false)))
+                .isSameAs(ETagResponse.SKIP_METHOD_NOT_MODIFIED);
+    }
+
+    /**
+     * Make sure that if-none-match always trumps if-modified-since.
+     */
+    @Test
+    void testIfNoneMatchIfModifiedSinceEvaluationOrderConditionalRequest() {
+        final RequestHeaders reqHeaders =
+                RequestHeaders.builder(HttpMethod.GET, "/")
+                              .add(IF_NONE_MATCH, "\"foobar\"")
+                              .addTimeMillis(IF_MODIFIED_SINCE, LONG_TIME_AGO)
+                              .build();
+        assertThat(conditionalRequest(reqHeaders, new ETag("foobar", true), NOT_SO_LONG_TIME_AGO))
+                .isSameAs(ETagResponse.SKIP_METHOD_NOT_MODIFIED);
+        assertThat(conditionalRequest(reqHeaders, new ETag("foo", true), LONG_TIME_AGO))
+                .isSameAs(ETagResponse.PERFORM_METHOD);
```

Comment:
LONG_TIME_AGO on both sides - should imply not modified

```suggestion
                .isSameAs(ETagResponse.SKIP_METHOD_NOT_MODIFIED);
```